internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -1533,13 +1583,26 @@ func TestMachineSetReconciler_reconcileUnhealthyMachines(t *testing.T) {
 
 		_, err := r.reconcileUnhealthyMachines(ctx, s)
 		g.Expect(err).ToNot(HaveOccurred())
-		// Verify the unhealthy machine is deleted.
+
+		// Verify the unhealthy machine is deleted (deletionTimestamp must be set).
 		m := &clusterv1.Machine{}
-		err = r.Client.Get(ctx, client.ObjectKeyFromObject(unhealthyMachine), m)
-		g.Expect(apierrors.IsNotFound(err)).To(BeTrue())
+		g.Expect(r.Client.Get(ctx, client.ObjectKeyFromObject(unhealthyMachine), m)).To(Succeed())
+		g.Expect(m.DeletionTimestamp.IsZero()).To(BeFalse())
+		conditions.IsTrue(m, clusterv1.MachineOwnerRemediatedCondition)
```

Comment:
shoud this be
```suggestion
		g.Expect(conditions.IsTrue(m, clusterv1.MachineOwnerRemediatedCondition)).To(BeTrue())
```

Reply:
Absolutely. Thx, nice catch!
(fixed both occurences of this)